src/middlewared/middlewared/plugins/chart_releases_linux/portals.py

```diff
@@ -42,45 +48,68 @@ def retrieve_portals_for_chart_release(self, release_data, node_ip):
 
     @private
     def retrieve_portals_for_chart_release_impl(self, release_data, node_ip):
+        cleaned_portals = {}
         questions_yaml_path = os.path.join(
             release_data['path'], 'charts', release_data['chart_metadata']['version'], 'questions.yaml'
         )
         if not os.path.exists(questions_yaml_path):
-            return {}
+            return cleaned_portals
 
         if release_data['chart_metadata']['name'] == 'ix-chart':
-            return self.get_ix_chart_portal(release_data, node_ip)
+            cleaned_portals.update(self.get_ix_chart_portal(release_data, node_ip))
 
         with open(questions_yaml_path, 'r') as f:
             portals = yaml.safe_load(f.read()).get('portals') or {}
 
-        if not portals:
-            return portals
-
         def tag_func(key):
             return self.parse_tag(release_data, key, node_ip)
 
-        cleaned_portals = {}
         for portal_type, schema in portals.items():
             t_portals = []
             path = tag_func(schema.get('path') or '/')
             for protocol in filter(bool, map(tag_func, schema['protocols'])):
                 for host in filter(bool, map(tag_func, schema['host'])):
                     for port in filter(bool, map(tag_func, schema['ports'])):
-                        t_portals.append(f'{protocol}://{host}:{port}{path}')
+                        t_portals.append(f'{protocol}://{host}{normalized_port_value(protocol, port)}{path}')
 
             cleaned_portals[portal_type] = t_portals
 
+        cleaned_portals.update(self.get_user_configured_portals(release_data, node_ip))
         return cleaned_portals
 
+    @private
+    def get_user_configured_portals(self, release_data, node_ip):
+        portals = {}
+        custom_portals = release_data['config'].get(CUSTOM_PORTALS_KEY)
+        if custom_portals is None:
+            return portals
+        try:
+            json_schema_validate({CUSTOM_PORTALS_KEY: custom_portals}, IX_VALUES_JSON_SCHEMA)
+        except JsonValidationError:
+            return portals
+
+        for portal_config in release_data['config'].get(CUSTOM_PORTALS_KEY) or []:
+            path = portal_config.get('path') or ''
+            host = node_ip if portal_config['useNodeIP'] else portal_config['host']
```

Comment:
if host is not given, you can just default to the NodeIP... there is not really a reason for the useNodeIP option I think?

Reply:
We should be explicit where we can be and this helps a lot especially when user is filling the portal form. It would be very weird to say "Leave host empty and then Node IP will be used" as compared to having a Use Node IP checkbox which when unchecked shows the host field.

Reply:
We do that but that has it's drawbacks, node ip is something which can change when user changes it and then we can't change that value in each app's configuration. So it's simpler to just let them specify if they want whatever is the node ip atm to be used :)